hoot-test/src/main/cpp/hoot/test/main.cpp

```diff
@@ -205,21 +205,76 @@ void printNames(CppUnit::Test* t)
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry().makeTest());
+    suite->addTest(new ScriptTestSuite("test-files/cmd/current/", printDiff));
+    suite->addTest(new ScriptTestSuite("test-files/cmd/quick/", printDiff));
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("current").makeTest());
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("quick").makeTest());
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("TgsTest").makeTest());
+    break;
+  case QUICK_ONLY:
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry().makeTest());
+    suite->addTest(new ScriptTestSuite("test-files/cmd/quick/", printDiff));
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("quick").makeTest());
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("TgsTest").makeTest());
+    break;
+  case SLOW:
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry().makeTest());
+    suite->addTest(new ScriptTestSuite("test-files/cmd/current/", printDiff));
+    suite->addTest(new ScriptTestSuite("test-files/cmd/quick/", printDiff));
+    suite->addTest(new ScriptTestSuite("test-files/cmd/slow/", printDiff));
+    suite->addTest(new ConflateCaseTestSuite("test-files/cases"));
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("current").makeTest());
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("quick").makeTest());
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("TgsTest").makeTest());
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("slow").makeTest());
+    break;
+  case SLOW_ONLY:
+    suite->addTest(new ScriptTestSuite("test-files/cmd/slow/", printDiff));
+    suite->addTest(new ConflateCaseTestSuite("test-files/cases"));
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("slow").makeTest());
+    break;
+  case GLACIAL:
+  case ALL:
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry().makeTest());
+    suite->addTest(new ScriptTestSuite("test-files/cmd/current/", printDiff));
+    suite->addTest(new ScriptTestSuite("test-files/cmd/quick/", printDiff));
+    suite->addTest(new ScriptTestSuite("test-files/cmd/slow/", printDiff));
+    suite->addTest(new ScriptTestSuite("test-files/cmd/glacial/", printDiff));
+    suite->addTest(new ConflateCaseTestSuite("test-files/cases"));
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("current").makeTest());
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("quick").makeTest());
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("TgsTest").makeTest());
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("slow").makeTest());
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("glacial").makeTest());
+    break;
+  case GLACIAL_ONLY:
+    suite->addTest(new ScriptTestSuite("test-files/cmd/glacial/", printDiff));
+    suite->addTest(CppUnit::TestFactoryRegistry::getRegistry("glacial").makeTest());
+    break;
+  }
```

Comment:
@bmarchant - This structure looks good. It may be verbose, but it's very readable, and I think that's important. My suggestion is to add a comment to the top of this function, pointing out that the order is important, and maybe give a hint as to what the dependency is (i.e.: this must run before that, because...)